### ISSUE TITLE
icu4c lost on macos-14-arm64 in version 20241022.361

### ISSUE THREAD

**Issue by andreasabel** [actions/runner-images#10864]
### Description

Looks like the ICU library (package`icu4c`) got removed from macos-14-arm64 somewhere between 20241007.259 and 20241022.361.  The symptom is visible at https://github.com/agda/agda/actions/runs/11530774344/job/32100729494#step:7:11
> Package icu-i18n was not found in the pkg-config search path.

It was still present in 20241007.259, see https://github.com/agda/agda/actions/runs/11401211472/job/31723660531#step:1:8
>  Image: macos-14-arm64
>  Version: 20241007.259

It is absent in 20241022.361, see https://github.com/agda/agda/actions/runs/11530774344/job/32100729494#step:1:8
>  Image: macos-14-arm64
>  Version: 20241022.361

Workaround is `brew install icu4c`, workflow run https://github.com/agda/agda/actions/runs/11531272325/job/32101789092#step:1:8
>  Image: macos-14-arm64
>  Version: 20241022.361

Not sure why packages get removed from one minor release to another.  After all, as a user of these runners I cannot request a specific version.  I can just ask for major versions like `macos-14` etc.  It would be acceptable to remove packages for major bumps, e.g. going from `macos-14` to `macos-15`.  But just dropping stuff (without offering the possibility to pin to a certain version) inevitably results in breakages downstream.

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [X] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://github.com/agda/agda/actions/runs/11530774344/job/32100729494#step:7:11

### Is it regression?

Yes, succeeds in 20241007.259 see description.

### Expected behavior

icu4c installed on runner per default

### Actual behavior

icu4c no longer installed

### Repro steps

See description with linked workflows.

**Comment by sureshe456** [comment]
Hi @andreasabel We are looking into the issue.Will keep you posted with updates.

**Comment by susmitamane** [comment]
Hi @andreasabel Nothing changed for macOS images, there is update from homebrew site for package path. They have updated different paths for intel and ARM64 architecture.

Previously, you could set the PKG_CONFIG_PATH to /usr/local/opt/icu4c/lib/pkgconfig, and it would work for both Intel and Apple Silicon systems (if you were running the same architecture). But with the new architecture separation in Homebrew, Apple Silicon requires you to update the PKG_CONFIG_PATH to use /opt/homebrew/ instead of /usr/local/.

If you want to use both images then please refer below. anyhow you have already performed workaround in your script. 
```
if [[ "$(uname -m)" == "arm64" ]]; then
  # Apple Silicon (M1/M2)
  export PKG_CONFIG_PATH="/opt/homebrew/opt/icu4c/lib/pkgconfig:$PKG_CONFIG_PATH"
else
  # Intel-based Mac
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:$PKG_CONFIG_PATH"
fi

```
<img width="1403" alt="Screenshot 2024-11-15 at 3 31 49 PM" src="https://github.com/user-attachments/assets/2cc140fc-da9b-4fb3-a7da-0f7b4c55d112">

**Comment by andreasabel** [comment]
@susmitamane:  But I already used the flexible setup of `PKG_CONFIG_PATH` that gives me the correct path for ARM and Intel:
https://github.com/agda/agda/actions/runs/11530774344/job/32100729494#step:6:2
```
PKG_CONFIG_PATH=$(brew --prefix)/opt/icu4c/lib/pkgconfig
```

Anyhow, I think you fixed the problem in macos-14-arm64 20241108.442 https://github.com/agda/agda/actions/runs/11831620001/job/32967077629#step:1:9 .
Because there I see now for `brew install icu4c` https://github.com/agda/agda/actions/runs/11831620001/job/32967077629#step:6:17
```
icu4c@76 76.1_1 is already installed and up-to-date. To reinstall 76.1_1, run: brew reinstall icu4c@76
```

**Comment by susmitamane** [comment]
Yeah, Thanks for the confirmation. closing this issue.

Feel free to get back to us for mac-os related issues. Thank you.
